core/quorum/vote_strategy.go

```diff
@@ -1,99 +1,156 @@
+		return []byte(`"inactive"`), nil
+	}
+	return nil, fmt.Errorf("Could not determine status")
+}
+
+type BlockVoteMakerStrategy interface {
+	// Start generating blocks/voting
 	Start() error
 	// (temporary) stop generating blocks
-	Pause() error
-	// Resume after a pause
-	Resume() error
+	PauseBlockMaking() error
+	// Resume block maker after a pause
+	ResumeBlockMaking() error
+	// (temporary) stop generating voting events
+	PauseVoting() error
+	// Resume voting after a pause
+	ResumeVoting() error
 	// Status returns indication if this implementation
-	// is generation CreateBlock events.
-	Status() BlockMakerStatus
+	// is generation CreateBlock and/or Voting events.
+	Status() (Status, Status)
 }
 
 // randomDeadlineStrategy asks the block voter to generate blocks
 // after a deadline is passed without importing a new head. This
 // deadline is chosen random between 2 limits.
 type randomDeadlineStrategy struct {
-	mux           *event.TypeMux
-	min, max      int // min and max deadline
-	activeMu      sync.Mutex
-	active        bool
-	deadlineTimer *time.Timer
+	mux                        *event.TypeMux
+	minBlockTime, maxBlockTime int // min and max block creation deadline
+	minVoteTime, maxVoteTime   int // min and max block voting deadline
+	activeMu                   sync.Mutex
+	blockCreateActive          bool
+	votingActive               bool
+	voteTimer                  *time.Timer
+	deadlineTimer              *time.Timer
+	rand                       *rand.Rand
 }
 
 // NewRandomDeadelineStrategy returns a block maker strategy that
 // generated blocks randomly between the given min and max seconds.
-func NewRandomDeadelineStrategy(mux *event.TypeMux, min, max uint) *randomDeadlineStrategy {
-	if min > max {
-		min, max = max, min
+func NewRandomDeadelineStrategy(mux *event.TypeMux, minBlockTime, maxBlockTime, minVoteTime, maxVoteTime uint) *randomDeadlineStrategy {
+	if minBlockTime > maxBlockTime {
+		minBlockTime, maxBlockTime = maxBlockTime, minBlockTime
 	}
-	if min == 0 {
+	if minBlockTime == 0 {
 		glog.Info("Set minimum block deadeline interval to 1 second")
-		min += 1
+		minBlockTime += 1
+	}
+	if minBlockTime == maxBlockTime {
+		maxBlockTime += 1
 	}
-	if min == max {
-		max += 1
+
+	if minVoteTime > maxVoteTime {
+		minVoteTime, maxVoteTime = maxVoteTime, minVoteTime
 	}
+	if minVoteTime == 0 {
+		glog.Info("Set minimum block deadeline interval to 1 second")
+		minVoteTime += 1
+	}
+	if minVoteTime == maxVoteTime {
+		maxVoteTime += 1
+	}
+
+	seed, err := crand.Int(crand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		glog.Fatalf("Unable to seed (psuedo) random deadline strategy")
+	}
+
 	s := &randomDeadlineStrategy{
-		mux:    mux,
-		min:    int(min),
-		max:    int(max),
-		active: true,
+		mux:               mux,
+		minBlockTime:      int(minBlockTime),
+		maxBlockTime:      int(maxBlockTime),
+		minVoteTime:       int(minVoteTime),
+		maxVoteTime:       int(maxVoteTime),
+		blockCreateActive: true,
```

Comment:
The `blockCreateActive` and `voteActive` flags are always set to `true`, can the default values be based on the role of the node? e.g: if the node is configured for voting, then `voteActive` will be `true`, otherwise `false`. I think the voting and block creation timers should only be activated based on the role of the node.